diff -Nru agata/agata.php agata.new/agata.php
--- agata/agata.php	Sat Mar 22 18:31:23 2003
+++ agata.new/agata.php	Sat Mar 22 18:41:10 2003
@@ -43,6 +43,7 @@
   var $aCheckVars;
   var $aAgataConfig;
   var $aLang;
+  var $aConfigDir;
 
   function AgataMain($aAgataConfig, $aDescription, $setupScreenLanguage)
   {
@@ -175,8 +176,15 @@
 
 
 //Gtk::rc_parse('./AquaGraphite/gtk/gtkrc');
-
-$aAgataConfig = parse_ini_file(getenv("HOME")."/.agata",TRUE);
+$aConfigDir = getenv("ETC_DIR");
+if ($aConfigDir)
+{
+  $aAgataConfig = parse_ini_file(getenv($aConfigDir . "/agata",TRUE);
+}
+else
+{
+  $aAgataConfig = parse_ini_file(getenv("HOME")."/.agata",TRUE);
+}
 $agapp = new AgataMain(&$aAgataConfig, $aDescription, $setupScreenLanguage);
 Gtk::main();
 ?>
diff -Nru agata/classes/AgataCore.class agata.new/classes/AgataCore.class
--- agata/classes/AgataCore.class	Sat Mar 22 18:31:23 2003
+++ agata.new/classes/AgataCore.class	Sat Mar 22 18:39:52 2003
@@ -143,8 +143,15 @@
       fwrite ($fd, "?>\n");
       fclose($fd);  
     }
-    
-    $fd = @fopen (getenv("HOME")."/.agata", "w");
+    $etc_dir = getenv("ETC_DIR");
+    if ($etc_dir)
+    {
+      $fd = @fopen ($etc_dir . "/agata", "w");
+    }
+    else
+    {
+      $fd = @fopen (getenv("HOME")."/.agata", "w");
+    }
     if ($fd)
     {
       if ($project=='+')
